src/Illuminate/Database/Schema/Grammars/SqlServerGrammar.php

```diff
@@ -875,6 +883,10 @@ public function wrapTable($table)
         if ($table instanceof Blueprint && $table->temporary) {
             $this->setTablePrefix('#');
         }
+        if(!$table  instanceof Blueprint){
+            // fix infinite loop when $tablePrefix = 'schema.';
+            return $this->wrapValue($table);
```

Comment:
Why would there be an infinite loop?

Reply:
I had infinte loop while unit testing.
every time it concatenates the `schema.` prefix 
and then in wrapSegment() it splits on the dot. if it has 2 or more segments it will call wrapTable() method and thus the loop starts over again.  => you can test by removing this line and running docker compose

Reply:
example exception my tests threw before adding this part:
```
Error : Maximum function nesting level of '256' reached, aborting!
 E:\git\laravel-framework\src\Illuminate\Database\Grammar.php:173
 E:\git\laravel-framework\src\Illuminate\Database\Grammar.php:38
 E:\git\laravel-framework\src\Illuminate\Database\Schema\Grammars\Grammar.php:216
 E:\git\laravel-framework\src\Illuminate\Database\Schema\Grammars\SqlServerGrammar.php:891
 E:\git\laravel-framework\src\Illuminate\Database\Grammar.php:99
 E:\git\laravel-framework\src\Illuminate\Collections\Collection.php:656
 E:\git\laravel-framework\src\Illuminate\Database\Grammar.php:101
 E:\git\laravel-framework\src\Illuminate\Database\Grammar.php:65
 E:\git\laravel-framework\src\Illuminate\Database\Schema\Grammars\Grammar.php:230
 E:\git\laravel-framework\src\Illuminate\Database\Grammar.php:39
 E:\git\laravel-framework\src\Illuminate\Database\Schema\Grammars\Grammar.php:216
 E:\git\laravel-framework\src\Illuminate\Database\Schema\Grammars\SqlServerGrammar.php:891
 E:\git\laravel-framework\src\Illuminate\Database\Grammar.php:99
 E:\git\laravel-framework\src\Illuminate\Collections\Collection.php:656
 E:\git\laravel-framework\src\Illuminate\Database\Grammar.php:101
 E:\git\laravel-framework\src\Illuminate\Database\Grammar.php:65
 E:\git\laravel-framework\src\Illuminate\Database\Schema\Grammars\Grammar.php:230
 E:\git\laravel-framework\src\Illuminate\Database\Grammar.php:39
 E:\git\laravel-framework\src\Illuminate\Database\Schema\Grammars\Grammar.php:216
 E:\git\laravel-framework\src\Illuminate\Database\Schema\Grammars\SqlServerGrammar.php:891
 E:\git\laravel-framework\src\Illuminate\Database\Grammar.php:99
```

Reply:
current setup does not allow `x.` prefix it will fall in `wrapSegments()` funciton thinking the first part is the table second part is column. the first table part will 'again' be prefixed and thus the loop.

Reply:
I still have no idea what we are talking about to be honest. It feels like we are hacking around some deeper issue. Please try explain why this happens very clearly with a walk through of the code with screenshots.